### ISSUE TITLE
SFDX version 7.155.0 is failing to create/install package version due to "Missing config object" error

### ISSUE THREAD

**Issue by honeyverma** [forcedotcom/cli#1576]
### Summary

We have our CI-CD pipeline which creates a new version for every PR/Push, however, suddenly it started failing with error: "Missing config object"
PS: We always installed a new version of SFDX in our pipeline server

### Steps To Reproduce: (in our unlocked package repo pipeline)

1. Install sfdx: `npm i sfdx-cli --global`
2. Dev-hub auth: `sfdx auth:jwt:grant --clientid *** --jwtkeyfile server.key --username *** --setdefaultdevhubusername -a <devHubAlias>`
3. Create package version: `sfdx force:package:version:create -p onagsf -d force-app -k test1234 --wait 120 -v OnProd2.0 -c --definitionfile config/project-scratch-def.json`

### Expected result
Version should be created (as it was created for last 1 year until today morning)

### Actual result
Failed with an error message: `ERROR running force:package:version:create:  Missing config object.`

### System Information

1. We run our CI-CD on GitHub actions
2. We are using `npm i sfdx-cli --global`
3. Version would: 7.155.0 (which released today)

### Additional information
SFDX release: [7.155.0](https://github.com/forcedotcom/cli/blob/main/releasenotes/sfdx/README.md#71550-june-16-2022-stable)

**Comment by honeyverma** [comment]
We found the same issue for installation:
```
Run sfdx force:package:install -p "onagsf@4.11.0-47" -u OnUat2.0 -k test1234 -w 20
ERROR running force:package:install:  Missing config object.
```

**Comment by honeyverma** [comment]
It seems `npm i sfdx-cli --global` is installing `7.155.1`, which is the latest as of now.
As a workaround, I downgraded it in our CI-CD to `7.152.0`

**Comment by shetzel** [comment]
Thanks for posting.  Are you able to reproduce this locally (outside of CI)?  Are you able to reproduce this error with any other commands in the same plugin, e.g., `force:mdapi:legacy:convert`.  Can you please post just the stack trace output when you append `--dev-debug`

**Comment by honeyverma** [comment]
Yes, I installed the latest version 7.155.1 then I am got the same error locally, so far I tried with 2 commands:
1. `sfdx force:package:install -p onagsf@4.11.0-47 -u hUat2.0 -k test1234 -w 15 --dev-debug`

Result: 
```
...
sf:GlobalInfo INFO Writing to config file: /Users/honey.verma/.sf/sf.json +25ms
ERROR running force:package:install:  Missing config object. 
*** Internal Diagnostic ***

MissingConfigObject: Missing config object.
    at Function.wrap (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@salesforce/core/lib/sfError.js:61:25)
    at PackageInstallCommand.catch (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/salesforce-alm/dist/ToolbeltCommand.js:255:44)
    at async PackageInstallCommand._run (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@salesforce/command/lib/sfdxCommand.js:93:13)
    at async Config.runCommand (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@oclif/config/lib/config.js:173:24)
    at async SfdxMain.run (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@oclif/command/lib/main.js:27:9)
    at async SfdxMain._run (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@oclif/command/lib/command.js:43:20)
    at async Object.run (/opt/homebrew/lib/node_modules/sfdx-cli/dist/cli.js:162:47)
******
```


2. `sfdx force:package:version:create -p onagsf -d force-app -k test1234 --wait 60 -v hProd2.0 -c --definitionfile config/project-scratch-def.json --dev-debug`


Resullt:
```
...
  sf:GlobalInfo INFO Writing to config file: /Users/honey.verma/.sf/sf.json +29ms
ERROR running force:package:version:create:  Missing config object. 
*** Internal Diagnostic ***

MissingConfigObject: Missing config object.
    at Function.wrap (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@salesforce/core/lib/sfError.js:61:25)
    at PackageVersionCreateCommand.catch (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/salesforce-alm/dist/ToolbeltCommand.js:255:44)
    at async PackageVersionCreateCommand._run (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@salesforce/command/lib/sfdxCommand.js:93:13)
    at async Config.runCommand (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@oclif/config/lib/config.js:173:24)
    at async SfdxMain.run (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@oclif/command/lib/main.js:27:9)
    at async SfdxMain._run (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@oclif/command/lib/command.js:43:20)
    at async Object.run (/opt/homebrew/lib/node_modules/sfdx-cli/dist/cli.js:162:47)
******
```

 Btw, I tried to narrow down the issue, I tried 1st command (package installation) works fine on my scratch org, it seems it is failing for sandbox or production (devHub) only.

**Comment by honeyverma** [comment]
> Thanks for posting. Are you able to reproduce this locally (outside of CI)? Are you able to reproduce this error with any other commands in the same plugin, e.g., `force:mdapi:legacy:convert`. Can you please post just the stack trace output when you append `--dev-debug`

Yes, I installed the latest version 7.155.1 and I got the same error locally, so far I tried with 2 commands:
1. `sfdx force:package:install -p onagsf@4.11.0-47 -u hUat2.0 -k test1234 -w 15 --dev-debug`

Result: 
```
...
sf:GlobalInfo INFO Writing to config file: /Users/honey.verma/.sf/sf.json +25ms
ERROR running force:package:install:  Missing config object. 
*** Internal Diagnostic ***

MissingConfigObject: Missing config object.
    at Function.wrap (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@salesforce/core/lib/sfError.js:61:25)
    at PackageInstallCommand.catch (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/salesforce-alm/dist/ToolbeltCommand.js:255:44)
    at async PackageInstallCommand._run (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@salesforce/command/lib/sfdxCommand.js:93:13)
    at async Config.runCommand (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@oclif/config/lib/config.js:173:24)
    at async SfdxMain.run (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@oclif/command/lib/main.js:27:9)
    at async SfdxMain._run (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@oclif/command/lib/command.js:43:20)
    at async Object.run (/opt/homebrew/lib/node_modules/sfdx-cli/dist/cli.js:162:47)
******
```


2. `sfdx force:package:version:create -p onagsf -d force-app -k test1234 --wait 60 -v hProd2.0 -c --definitionfile config/project-scratch-def.json --dev-debug`


Resullt:
```
...
  sf:GlobalInfo INFO Writing to config file: /Users/honey.verma/.sf/sf.json +29ms
ERROR running force:package:version:create:  Missing config object. 
*** Internal Diagnostic ***

MissingConfigObject: Missing config object.
    at Function.wrap (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@salesforce/core/lib/sfError.js:61:25)
    at PackageVersionCreateCommand.catch (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/salesforce-alm/dist/ToolbeltCommand.js:255:44)
    at async PackageVersionCreateCommand._run (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@salesforce/command/lib/sfdxCommand.js:93:13)
    at async Config.runCommand (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@oclif/config/lib/config.js:173:24)
    at async SfdxMain.run (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@oclif/command/lib/main.js:27:9)
    at async SfdxMain._run (/opt/homebrew/lib/node_modules/sfdx-cli/node_modules/@oclif/command/lib/command.js:43:20)
    at async Object.run (/opt/homebrew/lib/node_modules/sfdx-cli/dist/cli.js:162:47)
******
```

 Btw, I tried to narrow down the issue, I tried 1st command (package installation) works fine on my scratch org, it seems it is failing for sandbox or production (devHub) only.

**Comment by shetzel** [comment]
When it worked for your scratch org, was it set as the default org for your project so you did not specify a targetusername, e.g., `-u hUat2.0`?  I'm wondering if this is related to another bug I'm fixing.  If you set your sandbox or prod org as the defaultusername for your project and try the package install or version create commands without specifying the targetusername  I'm curious if it works.

**Comment by honeyverma** [comment]
Hey @shetzel 

I provided the targetusername like for scratch org:
 `sfdx force:package:install -p onagsf@4.11.0-47 -u hBaseTest -k test1234 -w 15`
 
 and it worked fine for me

**Comment by honeyverma** [comment]
btw, I am facing the same issue with `sfdx force:package:version:list -p onagsf` where I just wanted to see the list.

Unfortunately, I need to downgrade the sfdx cli version bcz I couldn't move forward with development, so I couldn't test more on 7.155.1

**Comment by shetzel** [comment]
@honeyverma  - if you are able to, can you retry your steps after:
`sfdx plugins:install salesforce-alm@latest`
and let me know if that fixed it for you?

When finished testing uninstall that plugin version with:
`sfdx plugins:uninstall salesforce-alm`

**Comment by omazhar** [comment]
@honeyverma @shetzel 

We've been experiencing the same issue since 7.155.1 was released.

I just now tried installing the salesforce-alm plugin and somehow magically, the force:package:install command works.

I hope this was just to help troubleshoot as it looks like the salesforce-ulm plugin is archived.

**Comment by honeyverma** [comment]
Thanks @omazhar ,

@shetzel it worked for me as well, I uninstalled and downgraded it again.

**Comment by omazhar** [comment]
@shetzel Can this be fixed in 7.156.x?

**Comment by shetzel** [comment]
@omazhar @honeyverma - this should now be fixed in CLI v7.156.1, which is the latest production version promoted today.
